src/visitors/rules/oas2-structural/validateOAS2Schema.js

```diff
@@ -9,114 +9,184 @@ class ValidateOAS2Schema {
+        if (node && node.maximum && typeof node.maximum !== 'number') {
+          return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('number'), 'value');
+        }
         return null;
       },
       exclusiveMaximum(node, ctx) {
-        if (node && node.exclusiveMaximum && typeof node.exclusiveMaximum !== 'boolean') return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('boolean'), 'value');
+        if (node && node.exclusiveMaximum && typeof node.exclusiveMaximum !== 'boolean') {
+          return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('boolean'), 'value');
+        }
         return null;
       },
       minimum(node, ctx) {
-        if (node && node.minimum && typeof node.minimum !== 'number') return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('number'), 'value');
+        if (node && node.minimum && typeof node.minimum !== 'number') {
+          return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('number'), 'value');
+        }
         return null;
       },
       exclusiveMinimum(node, ctx) {
-        if (node && node.exclusiveMinimum && typeof node.exclusiveMinimum !== 'boolean') return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('boolean'), 'value');
+        if (node && node.exclusiveMinimum && typeof node.exclusiveMinimum !== 'boolean') {
+          return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('boolean'), 'value');
+        }
         return null;
       },
       maxLength(node, ctx) {
         if (node && node.maxLength) {
-          if (typeof node.maxLength !== 'number') return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('number'), 'value');
-          if (node.maxLength < 0) return ctx.createError('Value of maxLength must be greater or equal to zero.', 'value');
+          if (typeof node.maxLength !== 'number') {
+            return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('number'), 'value');
+          }
+          if (node.maxLength < 0) {
+            return ctx.createError('Value of maxLength must be greater or equal to zero.', 'value');
+          }
         }
         return null;
       },
       minLength(node, ctx) {
         if (node && node.minLength) {
-          if (typeof node.minLength !== 'number') return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('number'), 'value');
-          if (node.minLength < 0) return ctx.createError('Value of minLength must be greater or equal to zero.', 'value');
+          if (typeof node.minLength !== 'number') {
+            return ctx.createError(ctx.messageHelpers.fieldTypeMismatchMessageHelper('number'), 'value');
+          }
+          if (node.minLength < 0) {
+            return ctx.createError('Value of minLength must be greater or equal to zero.', 'value');
```

Comment:
```suggestion
            return ctx.createError('Value of minLength must be greater than or equal to zero.', 'value');
```

I'm feeling we could reuse this message with a placeholder for the key name.